src/software/thunderscope/gl/gl_widget.py

```diff
@@ -0,0 +1,352 @@
+        """Add a layer to this GLWidget
+        
+        :param layer: The GLLayer 
+        :param visible: Whether the layer is visible on startup
+
+        """
+        self.layers.append(layer)
+
+        if not visible:
+            layer.hide()
+
+        # Add the layer to the Layer menu
+
+        # Not using a checkable QAction in order to prevent menu from closing
+        # when an action is pressed
+        layer_checkbox = QCheckBox(layer.name, self.layers_menu)
+        layer_checkbox.setStyleSheet("QCheckBox { padding: 0px 8px; }")
+        layer_checkbox.setChecked(layer.isVisible())
+        layer_checkbox.stateChanged.connect(
+            lambda: layer.setVisible(layer_checkbox.isChecked())
+        )
+
+        layer_action = QWidgetAction(self.layers_menu)
+        layer_action.setDefaultWidget(layer_checkbox)
+
+        self.layers_menu_actions[layer.name] = layer_action
+        self.layers_menu.addAction(layer_action)
+
+    def remove_layer(self, layer: GLLayer):
+        """Remove a layer from this GLWidget
+        
+        :param layer: The GLLayer to remove
+
+        """
+        # Remove all graphics provided by this layer from the scene
+        graphics = [
+            graphic
+            for graphics_list in layer.graphics_list.graphics.values()
+            for graphic in graphics_list
+        ]
+        for graphic in graphics:
+            self.gl_view_widget.removeItem(graphic)
+
+        # Remove the layer
+        self.layers.remove(layer)
+
+        # Remove the layer from the Layer menu
+        layer_action = self.layers_menu_actions[layer.name]
+        self.layers_menu.removeAction(layer_action)
+
+    def refresh(self):
+        """Trigger an update on all the layers, adding/removing GLGraphicsItem 
+        returned by the layers to/from the GLViewWidget scene
+        """
+        if self.player:
+            self.replay_controls.refresh()
+
+        # Prevents RuntimeError: wrapped C/C++ object of type ___ has been deleted
+        # See: https://stackoverflow.com/a/60700622/20199855
+        if self.isVisible() == False:
+            return
+
+        for layer in self.layers:
+            added_graphics, removed_graphics = layer.refresh_graphics()
+
+            for added_graphic in added_graphics:
+                self.gl_view_widget.addItem(added_graphic)
+
+            for removed_graphic in removed_graphics:
+                self.gl_view_widget.removeItem(removed_graphic)
+
+    def set_camera_view(self, camera_view: CameraView):
+        """Set the camera position to a preset camera view
+
+        :param camera_view: the preset camera view
+
+        """
+        self.gl_view_widget.reset()
+        if camera_view == CameraView.ORTHOGRAPHIC:
+            self.gl_view_widget.setCameraPosition(
+                pos=pg.Vector(0, 0, 0), distance=1100, elevation=90, azimuth=-90
+            )
+            self.gl_view_widget.setCameraParams(fov=1.0)
```

Comment:
I noticed that I cant pan around while in orthographic view, unless I slightly change the angle/`elevation` of the camera to not be 90. Do you know why this is?

Reply:
Not entirely sure. If you hold ctrl + left click drag panning works, but middle click drag doesn't

Reply:
Oh okay, you're right. I didn't know I could also cntrl + drag